R/vroom_fwf.R

```diff
@@ -167,25 +167,45 @@ fwf_positions <- function(start, end = NULL, col_names = NULL) {
 
 #' @rdname vroom_fwf
 #' @export
-#' @param ... If the first element is a data frame,
-#'   then it must have all numeric columns and either one or two rows.
-#'   The column names are the variable names. The column values are the
-#'   variable widths if a length one vector, and if length two, variable start and end
-#'   positions. The elements of `...` are used to construct a data frame
-#'   with or or two rows as above.
+#' @param ... Named or unnamed arguments, each addressing one column. Each input
+#' should be either a single integer (a column width) or a pair of integers
+#' (column start and end positions). All arguments must have the same shape,
+#' i.e. all widths or all positions.
 fwf_cols <- function(...) {
   x <- lapply(list(...), as.integer)
+
```

Comment:
I actually cleaned up the logic in this function, updated the docs to match what actually happens (previously they suggested you could provide a data frame, but that seems to have never been true), and improved/updated the error messages.